### DISCUSSION TITLE
Could we add customized headers dynamically for OtlpMetricExporter?

### DISCUSSION THREAD

**Discussion by aaron-ai** [open-telemetry/opentelemetry-dotnet#4220] [Q&A] [answered]
We have the need to add our own generated signature for auth when the metric data was exported. How could we reach that? Thank you!

**Comment by cijothomas** [comment, marked as answer]
You can control the httpclient used by OTLPExporters, so that can be leveraged to achieve this goal.
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol#configure-httpclient
